app/views/nodes/modals/_add_node.html.erb

```diff
@@ -0,0 +1,97 @@
+<div
+  id="modal_add_<%= type %>_node"
+  class="modal hide"
+  tabindex="-1"
+  role="dialog"
+  aria-labelledby="modal_add_node_header"
+  aria-hidden="true"
+>
+  <div class="modal-header">
+    <button type="button" class="close" data-dismiss="modal" aria-hidden="true">
+      ×
+    </button>
+
+    <h3 id="modal_add_node_header">Add subnode</h3>
+  </div>
+
+  <div class="modal-body">
+    <% if type == :branch %>
+      <p>You are adding a new top-level node.</p>
+    <% else %>
+      <p>You are adding a child to the <strong><%= @node.label %></strong> node.</p>
+    <% end %>
+
+    <%# we need to scope the names of the radio buttons to this specific modal
+      # so that clicking a radio in the 'add child' modal won't deselect the
+      # radios in the 'add branch' modal %>
+    <div class="radio">
+      <label>
+        <%= radio_button_tag(
+          "add_#{type}_node",
+          :one,
+          true,
+          class: "add_node_radio") %>
+        Add one
+      </label>
+    </div>
+
+    <div class="radio">
+      <label>
+        <%= radio_button_tag(
+          "add_#{type}_node",
+          :multiple,
+          false,
+          class: "add_node_radio") %>
+        Add multiple
+      </label>
+    </div>
+
+    <hr>
+
+    <%= simple_form_for(
```

Comment:
I had a look at doing it with a single form and JS but this solution seemed much easier. I see the problem with the repetition of the `type` and `parent_id` inputs (could they be extracted to a helper method) but other than that I think having two separate forms can still make sense because "create a node" and "create multiple nodes" feel different enough from each other conceptually to justify having wholly separate forms (after all, they also have separate controller actions.)